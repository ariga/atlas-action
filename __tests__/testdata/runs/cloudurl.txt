{
  "Env": {
    "Driver": "mysql",
    "URL": {
      "Scheme": "mysql",
      "Opaque": "",
      "User": {},
      "Host": "localhost:3307",
      "Path": "/test",
      "RawPath": "",
      "ForceQuery": false,
      "RawQuery": "",
      "Fragment": "",
      "RawFragment": "",
      "DSN": "root:pass@tcp(localhost:3307)/test",
      "Schema": "test"
    },
    "Dir": "ent/migrate/migrations"
  },
  "Steps": [
    {
      "Name": "Migration Integrity Check",
      "Text": "File atlas.sum is valid"
    },
    {
      "Name": "Detect New Migration Files",
      "Text": "Found 2 new migration files (from 2 total)"
    },
    {
      "Name": "Replay Migration Files",
      "Text": "Loaded 2 changes on dev database"
    },
    {
      "Name": "Analyze 20220815111810_create_users.up.sql",
      "Text": "0 reports were found in analysis",
      "Result": {
        "Name": "20220815111810_create_users.up.sql",
        "Text": "-- create \"users\" table\nCREATE TABLE `users` (`id` bigint NOT NULL AUTO_INCREMENT, PRIMARY KEY (`id`)) CHARSET utf8mb4 COLLATE utf8mb4_bin;\n"
      }
    },
    {
      "Name": "Analyze 20220816130609_field.up.sql",
      "Text": "1 reports were found in analysis",
      "Result": {
        "Name": "20220816130609_field.up.sql",
        "Text": "-- modify \"users\" table\nALTER TABLE `users` ADD COLUMN `name` varchar(255) NOT NULL;\n",
        "Reports": [
          {
            "Text": "data dependent change detected",
            "Diagnostics": [
              {
                "Pos": 24,
                "Text": "Adding a non-nullable \"varchar\" column \"name\" on table \"users\" without a default value implicitly sets existing rows with \"\"",
                "Code": "MY101"
              }
            ]
          }
        ]
      }
    }
  ],
  "Schema": {
    "Current": "schema \"test\" {\n  charset = \"utf8mb4\"\n  collate = \"utf8mb4_0900_ai_ci\"\n}\n",
    "Desired": "table \"users\" {\n  schema  = schema.test\n  collate = \"utf8mb4_bin\"\n  column \"id\" {\n    null           = false\n    type           = bigint\n    auto_increment = true\n  }\n  column \"name\" {\n    null = false\n    type = varchar(255)\n  }\n  primary_key {\n    columns = [column.id]\n  }\n}\nschema \"test\" {\n  charset = \"utf8mb4\"\n  collate = \"utf8mb4_0900_ai_ci\"\n}\n"
  },
  "Files": [
    {
      "Name": "20220815111810_create_users.up.sql",
      "Text": "-- create \"users\" table\nCREATE TABLE `users` (`id` bigint NOT NULL AUTO_INCREMENT, PRIMARY KEY (`id`)) CHARSET utf8mb4 COLLATE utf8mb4_bin;\n"
    },
    {
      "Name": "20220816130609_field.up.sql",
      "Text": "-- modify \"users\" table\nALTER TABLE `users` ADD COLUMN `name` varchar(255) NOT NULL;\n",
      "Reports": [
        {
          "Text": "data dependent change detected",
          "Diagnostics": [
            {
              "Pos": 24,
              "Text": "Adding a non-nullable \"varchar\" column \"name\" on table \"users\" without a default value implicitly sets existing rows with \"\"",
              "Code": "MY101"
            }
          ]
        }
      ]
    }
  ]
}