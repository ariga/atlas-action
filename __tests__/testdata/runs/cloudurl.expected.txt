<h1>Atlas Lint Report</h1>

<table><tr><th>Status</th><th>Step</th><th>Result</th><th>Diagnostics</th></tr><tr><td><div align="center"><img src="https://release.ariga.io/images/assets/success.svg" /></div></td><td>Migration Integrity Check</td><td>File atlas.sum is valid</td><td></tr><tr><td><div align="center"><img src="https://release.ariga.io/images/assets/success.svg" /></div></td><td>Detect New Migration Files</td><td>Found 2 new migration files (from 2 total)</td><td></tr><tr><td><div align="center"><img src="https://release.ariga.io/images/assets/success.svg" /></div></td><td>Replay Migration Files</td><td>Loaded 2 changes on dev database</td><td></tr><tr><td><div align="center"><img src="https://release.ariga.io/images/assets/success.svg" /></div></td><td>Analyze 20220815111810_create_users.up.sql</td><td>0 reports were found in analysis</td><td></tr><tr><td><div align="center"><img src="https://release.ariga.io/images/assets/warning.svg" /></div></td><td>Analyze 20220816130609_field.up.sql</td><td>1 reports were found in analysis</td><td>Adding a non-nullable "varchar" column "name" on table "users" without a default value implicitly sets existing rows with "" (<a href="https://atlasgo.io/lint/analyzers#MY101">MY101</a>)</td></tr></table>
